index.js

```diff
@@ -38,8 +38,51 @@ alfy.alfred = {
 
 alfy.input = process.argv[2];
 
-alfy.output = arr => {
-	console.log(JSON.stringify({items: arr}, null, '\t'));
+const isDefined = x => x !== null && x !== undefined;
+
+const wrapArg = item => {
+	const alfredworkflow = {arg: item.arg, variables: item.variables};
+	const arg = JSON.stringify({alfredworkflow});
+	const newItem = Object.assign({}, item, {arg});
+	delete newItem.variables;
+	return newItem;
+};
+
+const formatMods = item => {
+	const copy = Object.assign({}, item);
+
+	Object.keys(item.mods).forEach(modKey => {
+		const modItem = item.mods[modKey];
+		copy.mods[modKey] = wrapArg(modItem);
+	});
+
+	return copy;
+};
+
+// see https://www.alfredforum.com/topic/9070-how-to-workflowenvironment-variables/
+// for documentation on setting environment variables from Alfred workflows
+const format = item => {
+	if (!isDefined(item)) {
```

Comment:
I tried outputting an array of `null`s and noticed something pretty interesting. If you're calling your script from a script filter, Alfred just hangs when you output an array of non-objects.

```js
// Example 1: Alfred hangs when this is called from a script filter
alfy.output(['hi', 'hey']);

// Example 2: same for this
alfy.output([null, null, undefined]);
```

When called from a Run Script action, you just get stringified JSON as your Alfred `{query}`:

```
'{
  "items": [
    null,
    null
  ]
}'
```

How does that influence our decision here? Not quite sure. If we use

```js
item = item || {};
```

that will protect users from example 2 above, but not from example 1. What do you think?